astrovascpy/scipy_petsc_conversions.py

```diff
@@ -407,23 +394,14 @@ def PETScVec2array(x):
     istart, iend = x.getOwnershipRange()
 
     nloc = iend - istart
-    Istart = MPI_COMM.gather(istart, root=0)
-    Iend = MPI_COMM.gather(iend, root=0)
+    Nloc = tuple(MPI_COMM.allgather(nloc))
```

Comment:
why allgather instead of just gather? 
The same applied for Istart.

Reply:
I changed it to a simple `gather` function.